src/crypto-api/index.ts

```diff
@@ -364,6 +365,22 @@ export interface CryptoApi {
      */
     getEncryptionInfoForEvent(event: MatrixEvent): Promise<EventEncryptionInfo | null>;
 
+    /**
+     * Encrypts a given payload object via Olm to-device messages to a given
+     * set of devices.
+     *
+     * @param eventType the type of the event to send
+     * @param devices an array of (user ID, device ID) pairs to encrypt the payload for
```

Comment:
I wonder if we can/should impart any advice about the maximum number of entries in `devices`?

Reply:
A cursory look suggests that the limiting factor would be the size of the claim request generated by `KeyClaimManager.ensureSessionsForUsers()`.

Might it make sense to batch that at some sensible level? Similar to the batch size of 20 that is imposed for actually sending the to-device events to the server.

As a user of the higher-level CryptoApi I think any batching would be done for me automatically (which currently isn't the case).

Reply:
Agreed. I've made https://github.com/matrix-org/matrix-js-sdk/issues/4468 for this.